### PR TITLE
#163519108 Implement filter by tags when viewing an article

### PR DESCRIPTION
**Pull request by mcaleb808**
#### What does this PR do?
Implement filter by tags when viewing an article

#### Description of Task to be completed?
- add tests
- add filter by tags in view article components

#### How should this be manually tested?
- git clone
- npm install
- npm run start
- Navigate to articles`base_url/articles`
- Click on any article presented
- In view one article click on any `Tag` to go to articles with similar tag

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
`#163519108`

### CODE REVIEW

**Review by Nziranziza — APPROVED** [review]
Good work @mcaleb808

**Comment by mcaleb808** [comment]
> Good work @mcaleb808

Thank you @Nziranziza

File: src/__tests__/actions/currentUserActions.test.js [hunk 1]
```diff
@@ -134,18 +144,18 @@ describe('currentUserActions', () => {
       const expectedActions = [
         {
           type: SET_CURRENT_USER_FOLLOWING,
-          payload: true,
+          payload: true
         },
         {
           type: SET_USER_ACTION_FAILURE,
-          payload: 'User not found',
```
Comment (kimotho-njoki) [inline src/__tests__/actions/currentUserActions.test.js]: I thought you guys were following the trailing commas system, what's happening here?

File: src/__tests__/components/ConfirmedEmailEmail.test.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  test('should render the <ConfirmedEmaiMessage  />', () => {
+    const renderedValue = renderer.create(<ConfirmedEmailMessage {...props} />).toJSON();
+    expect(renderedValue).toMatchSnapshot();
+  });
+
+  test('should render default state', () => {
+    wrapper = shallow(<ConfirmedEmailMessage {...props} />);
+    expect(wrapper.state()).toEqual(defaultState);
+  });
+  test('should render default state', () => {
```
Comment (kimotho-njoki) [inline src/__tests__/components/ConfirmedEmailEmail.test.js:39]: did we forget something?

File: src/__tests__/helpers/editorPlugins/addLink.test.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import addLinkPlugin from '../../../helpers/editorPlugins/addLink';
+
+describe('addLinkPlugin', () => {
+  const editorState = EditorState.createEmpty();
+  const setEditorState = jest.fn();
+  test('should not handle key command', () => {
+    const command = 'not-add-link';
+    const { handleKeyCommand } = addLinkPlugin;
+    expect(handleKeyCommand(command, editorState, { setEditorState })).toEqual('not-handled');
+  });
```
Comment (kimotho-njoki) [inline src/__tests__/helpers/editorPlugins/addLink.test.js:12]: ?????????

File: src/__tests__/helpers/editorPlugins/addLink.test.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        };
+      };
+      return {
+        getSelection,
+      };
+    };
+    const addLink = keyBindingFn(event, { getEditorState });
+    expect(addLink).toEqual(undefined);
+  });
+  test('should return undefined', () => {
```
Comment (kimotho-njoki) [inline src/__tests__/helpers/editorPlugins/addLink.test.js:40]: ?????????????? Make your code readable. Beautiful. Space it and arrange it. Find a system and stick to it. Attention to Detail. You cannot space out one test then leave the rest.

File: src/__tests__/helpers/mediaBlockRenderer.test.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+describe('MediaBlockRender', () => {
+  test('should return component and editable property', () => {
+    const block = {
+      getType: () => 'atomic',
+    };
+    const { component, editable } = mediaBlockRenderer(block);
+    expect(component).toBeDefined();
+    expect(editable).toBeFalsy();
+  });
+  test('should return null', () => {
```
Comment (kimotho-njoki) [inline src/__tests__/helpers/mediaBlockRenderer.test.js:16]: I wont even comment.

File: src/components/Article/Article.jsx [hunk 6]
```diff
@@ -3,24 +3,51 @@ import { connect } from 'react-redux';
...
+
   componentDidMount() {
     const {
       match: {
         params: { articleSlug },
       },
       getArticle,
     } = this.props;
+
+    console.log(this.props);
```
Comment (kimotho-njoki) [inline src/components/Article/Article.jsx]: 🤣 .. 😠 ATTENTION TO DETAIL. REVIEW YOUR OWN PR. @mcaleb808 @filschristian @Nziranziza You will again pay for this.

File: src/components/Article/Article.jsx [hunk 7]
```diff
@@ -98,6 +126,28 @@ export class Article extends Component {
     e.preventDefault();
   };
 
+  navigateToArticles = (e) => {
```
Comment (kimotho-njoki) [inline src/components/Article/Article.jsx]: no need for `( )`

File: src/__tests__/actions/currentUserActions.test.js (hunk 1 above)
Comment (mcaleb808) [reply src/__tests__/actions/currentUserActions.test.js]: We are still  following the trailing commas system, I started this branch over unmerged branch, I've just updated this file

File: src/__tests__/components/ConfirmedEmailEmail.test.js (hunk 2 above)
Comment (mcaleb808) [reply src/__tests__/components/ConfirmedEmailEmail.test.js:39]: Yes we did! I added space

File: src/__tests__/helpers/editorPlugins/addLink.test.js (hunk 3 above)
Comment (mcaleb808) [reply src/__tests__/helpers/editorPlugins/addLink.test.js:12]: space added

File: src/__tests__/helpers/editorPlugins/addLink.test.js (hunk 4 above)
Comment (mcaleb808) [reply src/__tests__/helpers/editorPlugins/addLink.test.js:40]: space added

File: src/components/Article/Article.jsx (hunk 6 above)
Comment (mcaleb808) [reply src/components/Article/Article.jsx]: @kimotho-njoki I am guilty of this! I forget to remove `console.log`, It is removed now

File: src/components/Article/Article.jsx (hunk 7 above)
Comment (mcaleb808) [reply src/components/Article/Article.jsx]: `( )` removed

**Review by filschristian — APPROVED** [review]

**Comment by mcaleb808** [comment]
@kimotho-njoki I am fixing conflicts!

**Pull request merged**